frank-flow/src/frontend/src/app/editor/explorer/add-dialog/add-dialog.component.ts

```diff
@@ -28,8 +28,62 @@ export class AddDialogComponent {
     this.currentDirectory = this.codeService.currentDirectory;
   }
 
+  findDuplicateFileName(fileName: string, directory: any): boolean {
+    if (directory != null && fileName != null) {
+      const foundFile = directory._files.find(
+        (file: string) => file === fileName
+      );
+      if (foundFile) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  findCurrentDirectory(directory: any, path: string): any {
+    if (path === '') {
+      return directory;
+    } else {
+      let nextDirRegex = path.match(/.*(?=\/)/g);
+      let nextDir;
```

Comment:
You don't need to declare `nextDir` if you are going to make it in line >52 and >55.

Reply:
I do have to declare it because the variable would be out of scope when declared in an if check.